crates/storage/provider/src/writer/mod.rs

```diff
@@ -792,24 +807,22 @@ mod tests {
                 status: AccountStatus::Touched | AccountStatus::Created,
                 info: account_info.clone(),
                 storage: HashMap::default(),
+                transaction_id: 0,
             },
         )]));
-        state.merge_transitions(BundleRetention::Reverts);
-
-        // Block #7: Change storage.
         state.commit(HashMap::from_iter([(
             address1,
             RevmAccount {
-                status: AccountStatus::Touched,
+                status: AccountStatus::Touched | AccountStatus::SelfDestructed,
```

Comment:
Should this be reverted?

Reply:
I think so?

maybe this does actually explain the failing selfdestruct error?

Reply:
@klkvr we should add back 

before this commit:
```
        state.merge_transitions(BundleRetention::Reverts);

        // Block #7: Change storage.
 ```
 and after this commit:  
 ```
         state.merge_transitions(BundleRetention::Reverts);
```

Reply:
added this back and the test works again